torch/nn/modules/module.py

```diff
@@ -1384,7 +1358,20 @@ def state_dict(self, *args, destination=None, prefix='', keep_vars=False):
                 " and ".join(warn_msg) + " are deprecated. nn.Module.state_dict will not accept them in the future. "
                 "Refer to https://pytorch.org/docs/master/generated/torch.nn.Module.html#torch.nn.Module.state_dict for details.")
 
-        return self._state_dict_impl(destination, prefix, keep_vars)
+        destination._metadata[prefix[:-1]] = local_metadata = dict(version=self._version)
```

Comment:
I think accessing `destination._metadata` here will again cause the issue reported in #72778.